core/storage/filestore/file.hpp

```diff
@@ -48,17 +48,17 @@ namespace fc::storage::filestore {
      * @param buffer pointer to the first element to be read
      * @return number of bytes read
      */
-    virtual fc::outcome::result<size_t> read(size_t offset,
-                                             const gsl::span<uint8_t> &buffer) noexcept = 0;
+    virtual fc::outcome::result<size_t> read(
+        size_t offset, gsl::span<uint8_t> buffer) noexcept = 0;
 
     /**
      * @brief write to the file
      * @param offset start position to write to
      * @param buffer pointer to the first element of array to be written
      * @return number of bytes written
      */
-    virtual fc::outcome::result<size_t> write(size_t offset,
-                                              const gsl::span<const uint8_t> &buffer) noexcept = 0;
+    virtual fc::outcome::result<size_t> write(
+        size_t offset, gsl::span<const uint8_t> buffer) noexcept = 0;
 
     /**
```

Comment:
What's the rationale behind passing a span by value here?

Reply:
There is a convention in our projects to pass gsl::span by value rather than by reference.
gsl::span simply wraps two pointers, so it's cheap to copy.

Reply:
It was my fault.